### PR TITLE
 Support calling attached functions from attached functions in BIR

### PR DESCRIPTION
**Pull request by manuranga**
Sample:
```ballerina
type MyObj0 object {
   function a (int i){}
};
type MyObj1 object {
   function b (int i){
        MyObj0 myObj = new;
        myObj.a(8);
   }
};
function f(){
}
```

### CODE REVIEW

File: compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/main.bal [hunk 1]
```diff
@@ -30,10 +30,15 @@ public function main(string... args) {
     //do nothing
 }
 
-function generateJarBinary(bir:BIRContext birContext, bir:ModuleID entryModId, string progName) returns JarFile {
+function generateJarBinary(boolean dumpBir, bir:BIRContext birContext, bir:ModuleID entryModId, string progName) returns JarFile {
     currentBIRContext = birContext;
     bir:Package entryMod = birContext.lookupBIRModule(entryModId);
 
+    if(dumpBir) {
```
Comment (Kishanthan) [inline compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/main.bal]: not formatted

File: stdlib/bir/src/main/ballerina/bir/bir_emitter.bal [hunk 2]
```diff
@@ -338,9 +349,12 @@ type TypeEmitter object {
     function emitObjectType(BObjectType bObjectType, string tabs) {
         print(tabs, "object {");
         foreach var f in bObjectType.fields {
-            print(tabs + "\t", f.visibility, " ");
-            self.emitType(f.typeValue);
-            print(" ", f.name.value);
+            if(f is BObjectField){
```
Comment (Kishanthan) [inline stdlib/bir/src/main/ballerina/bir/bir_emitter.bal]: not formatted

Comment (manuranga) [reply stdlib/bir/src/main/ballerina/bir/bir_emitter.bal]: thanks will fix

**Comment by Kishanthan** [comment]
@manuranga , the streams related unit tests are failing with travis

File: stdlib/bir/src/main/ballerina/bir/bir_emitter.bal [hunk 3]
```diff
@@ -144,6 +152,10 @@ public type BirEmitter object {
         print("\t|\t");
         self.opEmitter.emitOp(errorEntry.errorOp);
     }
+
+    function emitVis(int i) {
+
```
Comment (warunalakshitha) [inline stdlib/bir/src/main/ballerina/bir/bir_emitter.bal]: Do we need this function. Cannot see any usages.

Comment (Kishanthan) [reply stdlib/bir/src/main/ballerina/bir/bir_emitter.bal]: yes, seems not used. shall we remove

**Comment by Kishanthan** [comment]
lets fix conflicts :)

**Comment by SupunS** [comment]
Fixed the conflicts and sent a separate PR: #14995

Hence closing this one.

**Pull request closed without merging**